### ISSUE TITLE
Nonnegative penalties actually allowed in CDRegressors

### ISSUE THREAD

**Issue by arose13** [scikit-learn-contrib/lightning#150]
The docs say that the allowed penalties are...
```python
penalty : str, 'l2', 'l1', 'l1/l2', 'nnl1', 'nnl2'
        The penalty to be used.

        - l2: ridge
        - l1: lasso
        - l1/l2: group lasso
        - nnl1: non-negative constraints + l1 penalty
        - nnl2: non-negative constraints + l2 penalty
```

But `nnl1` and `nnl2` do not work. This appears to be because the penalty object itself is not implemented. 

Unless there is some reason why this is not implemented in the Regressor? If not I'd love to contribute a PR for this.

**Comment by mblondel** [comment]
Indeed, the docstring seems wrong. Could you send a PR for correcting the docstring?

Support for non-negative constraints could be useful too but it seems to be a bit of work (you need to modify `primal_cd_fast.pyx`).

**Comment by mblondel** [comment]
`CDRegressor` doesn't use the penalty objects by the way.

If you need non-negative constraints, you can use [lbfgs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.fmin_l_bfgs_b.html) in scipy, which supports a `bounds` arguments.

**Comment by StrikerRUS** [comment]
Closed via #152.
